src/Microsoft.Health.Fhir.Api/Features/BackgroundJobService/HostingBackgroundService.cs

```diff
@@ -72,10 +74,12 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
                     await Task.Delay(TimeSpan.FromSeconds(1), cancellationTokenSource.Token);
                 }
 
-                foreach (var operation in _operationsConfiguration.HostingBackgroundServiceQueues)
+                var enabledQueueConfigs = GetEnabledQueueConfigs();
+
+                foreach (var queueConfig in enabledQueueConfigs)
```

Comment:
Look below at `OperationsConfiguration.cs`. We are combining the job hosting queue configuration with the job configuration as @brendankowitz suggested via inheritance.